src/oscar/apps/checkout/surcharges.py

```diff
@@ -0,0 +1,52 @@
+from decimal import Decimal as D
+
+from oscar.core import prices
+
+
+class Base(object):
+    """
+    Surcharge interface class
+
+    This is the superclass to the classes in surcharges.py. This allows using all
+    surcharges interchangeably (aka polymorphism).
+
+    The interface is all properties.
+    """
+
+    def calculate(self, basket, **kwargs):
+        raise NotImplementedError
+
+
+class PercentageCharge(Base):
+    name = "Percentage surcharge"
```

Comment:
I think this needs to be translatable?

Reply:
Given that names need to be translatable I think we also need a `code` here (same as we have for shipping methods) so that the charge can be uniquely identified.